### PR TITLE
 [feat] Add torchscript support for MMBT

### PR DESCRIPTION
**Pull request by vedanuj**
- Adds torchscript support for MMBT model
- Adds `BertEmbeddingsJit` and `BertModelJit` layers which are torchscriptable versions of `BertEmbeddings` and `BertModel` layers from transformers package
- Replaces transformers Bert model layers with scriptable layers by monkey patching, function `replace_with_jit`
- Adds tests

Test Plan:
- Run tests
- Run old pretrained checkpoints for MMBT and verify results

### CODE REVIEW

File: mmf/models/mmbt.py [hunk 1]
```diff
@@ -482,27 +473,42 @@ def __init__(self, config, *args, **kwargs):
...
         pooled_output = module_output[1]
         output = {}
 
-        if (
-            self.encoder_config.output_hidden_states
-            or self.encoder_config.output_attentions
-        ):
-            output["extras"] = module_output[2:]
+        if not torch.jit.is_scripting():
+            if (
```
Comment (apsdehal) [inline mmf/models/mmbt.py:503]: Some comments from other PR are also applicable here, please take care of those.

**Review by apsdehal — COMMENTED** [review]
Some of the comments from other PR are applicable here, please address those.

File: mmf/models/mmbt.py [hunk 2]
```diff
@@ -227,90 +230,70 @@ def forward(
...
-            encoder_extended_attention_mask = encoder_attention_mask[:, None, None, :]
-
-        encoder_extended_attention_mask = encoder_extended_attention_mask.to(
-            dtype=next(self.parameters()).dtype
-        )  # fp16 compatibility
-        encoder_extended_attention_mask = (
-            1.0 - encoder_extended_attention_mask
-        ) * -10000.0
-
-        # Prepare head mask if needed
```
Comment (vedanuj) [inline mmf/models/mmbt.py:276]: Removing this as we always set head_mask to None in the call to `MMBTModel`. Hence this is never used. Also scripting List[None] is not supported in the `BertEncoder`.

**Review by apsdehal — APPROVED** [review]
Probably need to check why CircleCI is having memory issues or disable that test for circleci cpu.

**Pull request closed without merging**